alacritty/src/config/window.rs

```diff
@@ -74,10 +74,24 @@ impl Default for WindowConfig {
 impl WindowConfig {
     #[inline]
     pub fn dimensions(&self) -> Option<Dimensions> {
-        if self.dimensions.columns.0 != 0
-            && self.dimensions.lines != 0
-            && self.startup_mode != StartupMode::Maximized
-        {
+        // Warn if one of columns, lines is non-zero but not both
+        if (self.dimensions.lines != 0) ^ (self.dimensions.columns.0 != 0) {
+            let (zero_key, key, value) = match self.dimensions.lines != 0 {
+                true => ("columns", "lines", self.dimensions.lines),
+                false => ("lines", "columns", self.dimensions.columns.0),
+            };
+
+            warn!(
+                target: LOG_TARGET_CONFIG,
+                "Both lines and columns must be non-zero for window.dimensions to take effect. \
+                Configured value of {} is 0 while that of {} is {}",
+                zero_key,
+                key,
+                value,
+            );
+        }
+
+        if (self.dimensions.lines != 0) && (self.dimensions.columns.0 != 0) {
```

Comment:
You check the same stuff 5 times. At that point it would be much easier to just assign it to a descriptive variable and check that instead.

It also is entirely pointless to check for warnings if we know that both dimensions and columns are `!= 0` anyway. So this should really be an `if -> else if -> else`.